### PR TITLE
[FX-5348] Update Avatar component to follow IMF guidelines

### PR DESCRIPTION
**Pull request by AdrianContiu**
[FX-5348]

### Description

This PR adds a new prop called `emblem` to the Avatar and AvatarGroup components so that the Toptal logo can be displayed if needed in the lower left corner. It also removes the `variant` prop that was used to display the avatar in a `portrait` or `landscape` view.

**_Note_**: The dimensions for the emblem are set according to the Figma [designs](https://www.figma.com/design/5SCTOPrCDcHuk5We091GBn/branch/bUFkua511v5UOOpn08KSZO/Product-Library?node-id=340-14433&t=Gj4q8N9lVNS4syvg-0).

### How to test

- [Temploy](https://picasso.toptal.net/fx-5348-update-avatar-component-to-follow-imf-guidelines)
- Check the [production](https://picasso.toptal.net/?path=/story/components-avatar--avatar) and compare it with the temploy.

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- ~Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)~
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- ~Annotate all `props` in component with documentation~
- [x] Create `examples` for component
- ~Ensure that deployed demo has expected results and good examples~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5348]: https://toptal-core.atlassian.net/browse/FX-5348?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by AdrianContiu** [comment]
@toptal-anvil ping reviewers

**Comment by TomasSlama** [comment]
Please add example of Avatar with emblem with different sizes as we have in the design, so we can easily compare

File: .changeset/polite-walls-invite.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@toptal/picasso-avatar': major
+---
+
+- add `emblem` prop to Avatar and AvatarGroup to enable display of Toptal logo
+- remove `variant` prop because we no longer display avatar as `portrait` or `landscape`
```
Comment (TomasSlama) [inline .changeset/polite-walls-invite.md:6]: lets also mention that the design have changed and maybe add link to the figma, so users can check how they should use it now.

File: packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx [hunk 2]
```diff
@@ -1,42 +1,80 @@
...
+  if (isTooSmall || !emblem) {
+    return false
+  }
+
+  return true
+}
+
+const AvatarLogo = ({ size }: Pick<Props, 'size'>) => (
+  <div
+    className={twJoin('flex absolute bottom-0', logoClassByPadding[size])}
```
Comment (TomasSlama) [inline packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx:34]: why `byPadding`? 🤔 maybe `bySize`

Comment (TomasSlama) [reply packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx:34]: I see, we already have logoClassBySize, so maybe we can rename to rootClassBySize.

Or we can have object with classes

const AvatarLogoClassesBySize= {
  root: {},
  logo: {},
}

File: .changeset/polite-walls-invite.md (hunk 1 above)
Comment (mkrl) [reply .changeset/polite-walls-invite.md:6]: According to our guidelines, a list of changed components should also be provided, for example:
```md
### Avatar, AvatarGroup
- [rest of the changes]
```

File: packages/base/Avatar/src/Avatar/story/ToptalLogo.example.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        src='./jacqueline-with-flowers-1954-square.jpg'
+        emblem
+      />
+    </Container>
+    <Container inline left={SPACING_4}>
+      <Avatar size='medium' name='Jacqueline Roque' emblem />
+    </Container>
+    <Container inline left={SPACING_4}>
+      <Avatar size='small' emblem />
+    </Container>
```
Comment (mkrl) [inline packages/base/Avatar/src/Avatar/story/ToptalLogo.example.tsx:21]:
```suggestion
    <Container inline left={SPACING_4}>
      <Avatar size='small' emblem />
    </Container>
    <Container inline left={SPACING_4}>
      <Avatar size='xsmall' emblem />
    </Container>
```

Just a "nice to have" suggestion, but I think it would be cool to also demo that on sizes below `small` the logo will not be visible

File: packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx [hunk 4]
```diff
@@ -1,42 +1,80 @@
...
+    role='img'
+    aria-label='photo placeholder'
+  >
+    <Logo
+      emblem
+      variant='blue'
+      // Necessary to decrease the default minWeight and minHeight so that the new width and height could apply
+      // eslint-disable-next-line no-inline-styles/no-inline-styles
+      style={{ minWidth: '1px', minHeight: '1px' }}
```
Comment (mkrl) [inline packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx:43]: To avoid inline styles, you can also use `base` prop for icons and set it to `1`, but unfortunately this prop is not passed down to icons in `@toptal/picasso-logo`. It could be added, but I'm honestly not sure if it's worth adding a new prop to a component v.s. having inline styles :smile: 
```suggestion
      base={1}
```

**Review by mkrl — APPROVED** [review]
Overall looks very accurate to the designs, nice job! Changeset is critical to be changed before merging (see comment from me and @TomasSlama), but the rest of the changes look good to me.

File: .changeset/polite-walls-invite.md (hunk 1 above)
Comment (AdrianContiu) [reply .changeset/polite-walls-invite.md:6]: There is a [discussion](https://toptal-core.slack.com/archives/GG3F4AS4T/p1715307550776449) on our channel where it has been decided that it is no longer necessary to provide the component title.

File: packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx (hunk 4 above)
Comment (AdrianContiu) [reply packages/base/Avatar/src/Avatar/AvatarWrapper/AvatarWrapper.tsx:43]: Yes, that was my first approach as well, but then I thought we want to have as few changes as possible, so I decided to go with the inline solution. If this is accepted then I prefer to go as it is.

File: packages/base/Avatar/src/Avatar/AvatarWrapper/styles.ts [hunk 5]
```diff
@@ -40,3 +20,20 @@ export const clipClassBySize: Record<Size, string> = {
   large:
     '[clip-path:polygon(0_0,_100%_0,_100%_100%,_1.5em_100%,_0_calc(100%-1.5em))]',
 }
+
+export const AvatarLogoClassesBySize: Record<string, Record<Size, string>> = {
```
Comment (ruslan-sed) [inline packages/base/Avatar/src/Avatar/AvatarWrapper/styles.ts:24]: Probably first letter should not be capitalized

File: packages/base/Avatar/src/AvatarGroup/AvatarGroup.tsx [hunk 6]
```diff
@@ -68,14 +71,15 @@ const AvatarGroup = ({
...
+            emblem={emblem}
             src={src}
             data-testid={avatarTestId}
           />
         )
       )}
       {isOverLimit ? (
         /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-        <AvatarWrapper size={size!} variant='square'>
+        <AvatarWrapper size={size!}>
```
Comment (ruslan-sed) [inline packages/base/Avatar/src/AvatarGroup/AvatarGroup.tsx:82]: I know that it is not related to your PR, sorry 😄 Is'nt non-null assertion error prone? Maybe we can omit it by providing a default value, wdyt?

Comment (ruslan-sed) [reply packages/base/Avatar/src/AvatarGroup/AvatarGroup.tsx:82]: And also for `Avatar`

**Review by ruslan-sed — COMMENTED** [review]
Good job, man 👍 
Left just few comments

File: packages/base/Avatar/src/Avatar/Avatar.tsx [hunk 7]
```diff
@@ -18,8 +18,8 @@ export interface Props extends StandardProps, HTMLAttributes<HTMLDivElement> {
   size?: SizeType<'xxsmall' | 'xsmall' | 'small' | 'medium' | 'large'>
   /** Photo url */
   src?: string
-  /** Variant of the avatar shape */
-  variant?: 'square' | 'portrait' | 'landscape'
+  /** Displays Toptal logo */
+  emblem?: boolean
```
Comment (TomasSlama) [inline packages/base/Avatar/src/Avatar/Avatar.tsx:22]: what do you think about variable name that would be clear that it is boolean value. `showEmblem`

**Review by TomasSlama — APPROVED** [review]

File: packages/base/Avatar/src/AvatarGroup/AvatarGroup.tsx (hunk 6 above)
Comment (AdrianContiu) [reply packages/base/Avatar/src/AvatarGroup/AvatarGroup.tsx:82]: Thanks for your comment, but as we were taught during the internship, never make changes that are not in the scope of PR. This change should happen in many places and because there could be a deeper implication of this that I don't know about and now it's too late to ask for a re-review since the sprint will end in a few hours, I decided not to proceed with it.  However, I think you can open the discussion during the daily and have a decision with the team.
